packages/desktop-client/src/util/ExposeNavigate.js

```diff
@@ -0,0 +1,10 @@
+import { useLayoutEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+export default function ExposeNavigate() {
+  let navigate = useNavigate();
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏why not use `useNavigate()` in all the individual components?

i.e.

```
const navigate = useNavigate();
navigate('/some/plage');
```

instead of 

```
window.__navigate('/some/place')
```

Reply:
All of the code paths that call `__navigate` fall into one of two categories:

1. Fully external code (e.g. Electron menu system) that can’t access the React tree in any way, but still has to trigger a navigation
2. Code that spans the router transition. For example, if you don’t have a server connected, clicking the “No server” button while a file is open will first close the file (causing a transition from the finance app to the management app and therefore a switch to the corresponding new router and an unmount of the component that triggered the transition), then navigate to `/change-server`. Calling `navigate` would likely not work correctly since it would be happening on a router other than the one currently controlling what’s displayed.